lib/db2.js

```diff
@@ -96,10 +110,12 @@ DB2.prototype.update = function(model, where, data, options, cb) {
  * @param {Function} cb The callback function
  */
 DB2.prototype.destroyAll = function(model, where, options, cb) {
+  debug('DB2.prototype.destroyAll: model=%s, where=%j, options=%j',
+        model, where, options);
   var self = this;
   var stmt = self.buildDelete(model, where, options);
-  var id = self.idName(model);
-  var sql = 'SELECT \"' + id + '\" FROM OLD TABLE (' + stmt.sql + ')';
+  var idName = self.idName(model);
+  var sql = 'SELECT \"' + idName + '\" FROM OLD TABLE (' + stmt.sql + ')';
```

Comment:
Actually I split the changes in two commits:
1) `Fix needed when id passed in` for the former changes
2) `Change id to idName` for the latter changes